src/http-api/utils.ts

```diff
@@ -79,17 +79,29 @@ export function parseErrorResponse(response: XMLHttpRequest | Response, body?: s
         return <Error>e;
     }
 
-    if (contentType?.type === "application/json" && body) {
-        return new MatrixError(
-            JSON.parse(body),
-            response.status,
-            isXhr(response) ? response.responseURL : response.url,
-        );
-    }
-    if (contentType?.type === "text/plain") {
-        return new HTTPError(`Server returned ${response.status} error: ${body}`, response.status);
-    }
-    return new HTTPError(`Server returned ${response.status} error`, response.status);
+    const httpHeaders = !isXhr(response)
```

Comment:
maybe move the call to `getResponseContentType` after this, so that it can operate on the resolved Headers value rather than re-engineering it?

Reply:
03af59df5